app/src/main/java/e/josephmolina/saywhat/MainScreen/MainController.java

```diff
@@ -73,15 +72,36 @@ public void startVoiceInput() {
         }
     }
 
-
     public void displaySpeechToTextResults(String results) {
         mainLayout.spokenText.setText(results);
         onTranslateClicked(results);
     }
 
     @Override
     public void onSpeakClicked(String text) {
-        textToSpeechManager.speak(text, TextToSpeech.QUEUE_FLUSH, null, null);
+
+        Completable.fromCallable((Callable<Void>) () -> {
+            PollyClient pollyClient = new PollyClient();
```

Comment:
Can this be initialized in the constructor of the MainController instead of reinitialized each time the speak button is clicked?

Reply:
yes it can. good call.